client/txn.go

```diff
@@ -384,47 +378,40 @@ func (txn *Txn) exec(retryable func(txn *Txn) error) error {
 // always commit or clean-up explicitly even when that may not be
 // required (or even erroneous).
 func (txn *Txn) send(calls ...proto.Call) error {
+	if txn.Proto.Status != proto.PENDING {
+		return util.Errorf("attempting to use %s transaction", txn.Proto.Status)
+	}
+
 	lastIndex := len(calls) - 1
 	if lastIndex < 0 {
 		return nil
 	}
 
+	lastReq := calls[lastIndex].Args
 	// haveTxnWrite tracks intention to write. This is in contrast to
 	// txn.Proto.Writing, which is set by the coordinator when the first
 	// intent has been created, and which lives for the life of the
 	// transaction.
-	var haveTxnWrite bool
-	var haveEndTxn bool
+	haveTxnWrite := proto.IsTransactionWrite(lastReq)
 
-	for i, call := range calls {
+	for _, call := range calls[:lastIndex] {
```

Comment:
> Yes, but that status is guaranteed to be reflected in txn.Proto.Writing before any EndTransaction call will be generated. So it matters, but it doesn't matter that it be included in `haveTxnWrite`.

Ah, you're right; I hadn't realized this. My only takeaway from this is that initializing `haveTxnWrite` to `proto.IsTransactionWrite(lastReq)` is useless, though.

> I'm just saying that either A) this loop should look at every request in the batch instead of dropping the last one (so that haveTxnWrite means haveTxnWrite) 

To be fair, we do look at the last one for the purposes of `haveTxnWrite`, we just do it first.

> or B) we need a comment explaining why it's ok to ignore the last request (wordsmith however you like).

Could you add this comment? I'm not really sure what you're looking for, since each request does get inspected for the purpose of `haveTxnWrite` (but not for the purpose of generating the non-terminal call error).